### PR TITLE
Add final destination annotation delegate method.

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

PR adds `NavigationMapViewDelegate.navigationMapView(_:didAdd:pointAnnotationManager)` to `CarPlayNavigationViewController` and `CarPlayMapViewController`.

Closing #3188.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -94,6 +94,8 @@
...
 * `RecentItem` is now a public struct. ([#2999](https://github.com/mapbox/mapbox-navigation-ios/pull/2999))
 * Removed unused `CarPlayManagerDelegate.carPlayManager(_:searchTemplate:updatedSearchText:ompletionHandler:)` and `CarPlayManagerDelegate.carPlayManager(_:carPlayManager:selectedResult:completionHandler:)`. ([#2999](https://github.com/mapbox/mapbox-navigation-ios/pull/2999))
+* Added the `CarPlayMapViewControllerDelegate.carPlayMapViewController(_:didAdd:pointAnnotationManager:)`, `CarPlayNavigationViewControllerDelegate.carPlayNavigationViewController(_:didAdd:pointAnnotationManager:)` and `CarPlayManager.carPlayManager(_:didAdd:to:pointAnnotationManager:)` delegate methods, which will be called whenever final destination `PointAnnotation` is added to `CarPlayMapViewController`, `CarPlayNavigationViewController` and `CarPlayManager` respectively. ([#3190](https://github.com/mapbox/mapbox-navigation-ios/pull/3190))
+* Added the `CarPlayMapViewControllerDelegate` public protocol, which provides methods for reacting to events during free-drive navigation or route previewing. ([#3190](https://github.com/mapbox/mapbox-navigation-ios/pull/3190))
```
Comment (1ec5) [inline CHANGELOG.md:98]: Nit: mention CarPlayMapViewControllerDelegate before its member:

```suggestion
* Added the `CarPlayMapViewControllerDelegate` public protocol, which provides methods for reacting to events during free-drive navigation or route previewing. ([#3190](https://github.com/mapbox/mapbox-navigation-ios/pull/3190))
* Added the `CarPlayMapViewControllerDelegate.carPlayMapViewController(_:didAdd:pointAnnotationManager:)`, `CarPlayNavigationViewControllerDelegate.carPlayNavigationViewController(_:didAdd:pointAnnotationManager:)` and `CarPlayManager.carPlayManager(_:didAdd:to:pointAnnotationManager:)` delegate methods, which will be called whenever the `PointAnnotation` representing the final destination is added to `CarPlayMapViewController`, `CarPlayNavigationViewController` and `CarPlayManager`, respectively. ([#3190](https://github.com/mapbox/mapbox-navigation-ios/pull/3190))
```

File: Sources/MapboxNavigation/CarPlayManagerDelegate.swift [hunk 2]
```diff
@@ -147,6 +148,19 @@ public protocol CarPlayManagerDelegate: AnyObject, UnimplementedLogging {
...
     func carPlayManager(_ carPlayManager: CarPlayManager, didPresent navigationViewController: CarPlayNavigationViewController) -> ()
+    
+    /**
+     Tells the receiver that the final destination `PointAnnotation` was added to either `CarPlayMapViewController` or `CarPlayNavigationViewController`.
+     
+     - parameter carPlayManager: The `CarPlayManager` object.
+     - parameter finalDestinationAnnotation: `PointAnnotation`, which was added to the `MapView`.
+     - parameter parentViewController: `UIViewController`, which contains `MapView` instance. Can be either `CarPlayMapViewController` or `CarPlayNavigationViewController`.
+     - parameter pointAnnotationManager: `PointAnnotationManager` instance, which is responsible for `PointAnnotation`s management in the `NavigationMapView`.
+     */
```
Comment (1ec5) [inline Sources/MapboxNavigation/CarPlayManagerDelegate.swift:159]: Nit: Sometimes it’s easier to read a method’s documentation if it uses fewer formal symbol references to types. These type references are also prone to getting out of sync during refactors, such as `NavigationMapView` versus `MapView`. The developer can still look at the method declaration to see what the types should be. That said, cross-references to related methods and types that aren’t obvious from the declaration are very much a good thing.

```suggestion
    /**
     Tells the receiver that the `PointAnnotation` representing the final destination was added to either `CarPlayMapViewController` or `CarPlayNavigationViewController`.
     
     - parameter carPlayManager: The `CarPlayManager` object.
     - parameter finalDestinationAnnotation: The point annotation that was added to the map view.
     - parameter parentViewController: The view controller that contains the map view, which is an instance of either `CarPlayMapViewController` or `CarPlayNavigationViewController`.
     - parameter pointAnnotationManager: The object that manages the point annotation in the map view.
     */
```

File: Sources/MapboxNavigation/CarPlayManagerDelegate.swift [hunk 3]
```diff
@@ -220,6 +234,14 @@ public extension CarPlayManagerDelegate {
         logUnimplemented(protocolType: CarPlayManagerDelegate.self, level: .debug)
     }
     
+    /**
+     `UnimplementedLogging` prints a warning to standard output the first time this method is called.
+     */
+    func carPlayManager(_ carPlayManager: CarPlayManager, shouldPresentArrivalUIFor waypoint: Waypoint) -> Bool {
```
Comment (1ec5) [inline Sources/MapboxNavigation/CarPlayManagerDelegate.swift:240]: I’m only now noticing that the existing delegate methods by this name would’ve been a good opportunity to use a preposition other than the overused “for”: `carPlayManager(_:shouldPresentArrivalUIFor:)`. 🤷‍♂️

**Review by 1ec5 — APPROVED** [review]

File: Sources/MapboxNavigation/CarPlayManagerDelegate.swift (hunk 2 above)
Comment (MaximAlien) [reply Sources/MapboxNavigation/CarPlayManagerDelegate.swift]: Yep, I guess using more generic naming can save some time in future when symbols are renamed.

**Pull request merged**